### PR TITLE
Build oci package

### PR DESCRIPTION
**Pull request by randomanderson**
**What does this PR do?**
Builds and publishes the OCI package for ruby

**Motivation:**
OCI packages are needed for the datadog install and fleet management

**How to test the change?**
Built packages need to be tested with internal tooling

### CODE REVIEW

**Review by BaptisteFoy — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -149,6 +149,55 @@ package-arm64:
...
+    - ../.gitlab/build-oci-package.sh
+
+oci-internal-publish:
+  extends: .oci-internal-publish
+  stage: package
+  needs: [ package-oci-arm64, package-oci-amd64 ]
+  rules:
+    - when: on_success
+  variables:
+    FLAVOR: datadog-apm-library-rb
```
Comment (BaptisteFoy) [inline .gitlab-ci.yml:185]:
```suggestion
    FLAVOR: datadog-apm-library-ruby
```

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -149,6 +149,55 @@ package-arm64:
...
+  needs: [ oci-internal-publish ]
+  rules:
+    - when: on_success
+  trigger:
+    project: DataDog/public-images
+    branch: main
+    strategy: depend
+  variables:
+    IMG_SOURCES: registry.ddbuild.io/ci/remote-updates/datadog-apm-library-rb:pipeline-${CI_PIPELINE_ID}-1
+    IMG_DESTINATIONS: apm-library-rb-package:pipeline-${CI_PIPELINE_ID}
```
Comment (BaptisteFoy) [inline .gitlab-ci.yml:198]:
```suggestion
    IMG_SOURCES: registry.ddbuild.io/ci/remote-updates/datadog-apm-library-ruby:pipeline-${CI_PIPELINE_ID}-1
    IMG_DESTINATIONS: apm-library-ruby-package:pipeline-${CI_PIPELINE_ID}
```

File: .gitlab/build-oci-package.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+mkdir sources
+
+cp -r ../tmp sources
+
+echo -n "$RUBY_PACKAGE_VERSION" > sources/version
+
+datadog-package create \
+  --version="$RUBY_PACKAGE_VERSION" \
+  --package="datadog-apm-library-rb" \
```
Comment (BaptisteFoy) [inline .gitlab/build-oci-package.sh:13]:
```suggestion
  --package="datadog-apm-library-ruby" \
```

File: .gitlab/build-oci-package.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  --archive=true \
+  --archive-path="datadog-apm-library-rb-$RUBY_PACKAGE_VERSION-$ARCH.tar" \
```
Comment (BaptisteFoy) [inline .gitlab/build-oci-package.sh:15]:
```suggestion
  --archive-path="datadog-apm-library-ruby-$RUBY_PACKAGE_VERSION-$ARCH.tar" \
```

File: .gitlab/build-oci-package.sh [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -e
+
+mkdir sources
+
+cp -r ../tmp sources
```
Comment (BaptisteFoy) [inline .gitlab/build-oci-package.sh:7]: **❓ Question:** ‏Shouldn't it be this instead?
```suggestion
cp -r ../tmp/* sources
```

Else the content of the package looks like this: 
```
$ ls -lah
total 16K
drwxr-xr-x 3 root root 4.0K May 27 11:31 .
drwxr-xr-x 3 root root 4.0K May 27 11:31 ..
drwxr-xr-x 6 root root 4.0K May 27 11:31 tmp
-rw-r--r-- 1 root root   41 May 27 11:31 version
```

**Review by BaptisteFoy — COMMENTED** [review]
Also I tested it, looks like https://github.com/DataDog/dd-trace-rb/blob/b908262754eed55fbca8826d130c870760716f07/lib-injection/host_inject.rb#L9 prevents us from using the right path

**Review by TonyCTHsu — CHANGES_REQUESTED** [review]
Generally, it is fine. 

I submitted `Request changes` because we need to validate this artifact before merging.

**Review by TonyCTHsu — APPROVED** [review]
The test is passing! 👍

**Pull request merged**